apps/prairielearn/src/lib/features/manager.ts

```diff
@@ -177,7 +178,30 @@ export class FeatureManager<FeatureName extends string> {
     await queryAsync(sql.disable_feature, { name, ...DEFAULT_CONTEXT, ...context });
   }
 
+  /**
+   * Runs the given function with a set of feature overrides persisted in
+   * {@link AsyncLocalStorage}. This allows for a feature to be enabled or
+   * disabled in the context of a single request.
+   */
   runWithOverrides<T>(overrides: FeatureOverrides, fn: () => T): T {
     return this.als.run(overrides, fn);
   }
+
+  /**
+   * Globally sets the given feature overrides and automatically cleans them up
+   * once the provided function has executed.
+   *
+   * Note that this should generally only be used in tests. If used while serving
+   * actual requests, this will have unintended behavior since the overrides will
+   * apply to all code that is running at the same time.
+   */
+  async runWithGlobalOverrides<T>(overrides: FeatureOverrides, fn: () => Promise<T>): Promise<T> {
```

Comment:
I ideally wanted to use `runWithOverrides` here, but in tests, overrides don't make their way down into the code that's actually handling the request. Hence, this _other_ way of doing overrides that doesn't rely on ALS.